syntax = "proto3";
package cmdb_service_ctrl;




import "google/protobuf/struct.proto";

/*
InstanceSearchAll请求
*/
message InstanceSearchAllRequest {
    /*
    模型对象ID
    */
    string objectId = 1 ;
    /*
    e.g.: { name: { $like: '%q%' } }, { $or: [{ name: { $like: '%q%' }}] }
    */
    google.protobuf.Struct query = 2 ;
    /*
    e.g.: { instanceId: true, name: true }
    */
    google.protobuf.Struct fields = 3 ;
    /*
    当为 true 时，只搜索与我相关实例
    */
    bool only_my_instance = 4 ;
    /*
    对于关联的实例数据是否只获取 relation_view 中指定的属性, 这个字段为 true 时, 会覆盖 fields 字段中指定的二级字段设置
    */
    bool only_relation_view = 5 ;
    /*
    按字段排序, 留空默认按照实例ID降序排序(1表示升序, -1表示降序) e.g.: { instanceId: 1 }
    */
    google.protobuf.Struct sort = 6 ;
    /*
    按照权限过滤(通用实例都有 read, update, delete 权限控制, 主机实例在通用实例权限基础上有额外的 operate 权限, 应用实例在通用实例权限基础上有额外的 developClusterOperate, testClusterOperate, prereleaseClusterOperate, productionClusterOperate 权限) e.g.: [ "operate", "update" ]
    */
    repeated string permission = 7 ;
}

/*
InstanceSearchAll返回
*/
message InstanceSearchAllResponse {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    错误信息
    */
    string error = 2 ;
    /*
    返回消息
    */
    string message = 3 ;
    /*
    实例列表
    */
    repeated google.protobuf.Struct data = 4 ;
}

/*
InstanceSearchAllApi返回
*/
message InstanceSearchAllResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    cmdb_service_ctrl.InstanceSearchAllResponse data = 4 ;
}

